tests/portForwardingServer.spec.ts

```diff
@@ -15,111 +15,134 @@
  */
 
 import http from 'http';
+import net from 'net';
 
-import { contextTest as it, expect } from './config/browserTest';
+import { PlaywrightClient } from '../lib/remote/playwrightClient';
+import { PlaywrightServer } from '../lib/remote/playwrightServer';
+
+import { contextTest, expect } from './config/browserTest';
 import type { LaunchOptions, ConnectOptions } from '../index';
+import type { Page, BrowserServer } from '..';
+
+type PageFactoryOptions = {
+  acceptForwardedPorts: boolean
+  forwardPorts: number[]
+};
+
+type LaunchMode = 'playwrightclient' | 'launchServer';
+
+const it = contextTest.extend<{ pageFactory: (options?: PageFactoryOptions) => Promise<Page> }, {launchMode: LaunchMode}>({
```

Comment:
It is fine like it is now, but `launchMode` might as well be test-scoped - you are not doing anything worker-scoped here.